prow/cmd/deck/main.go

```diff
@@ -581,7 +581,7 @@ func prodOnlyMain(cfg config.Getter, pluginAgent *plugins.ConfigAgent, o options
 	// When inrepoconfig is enabled, both the GitHubClient and the gitClient are used to resolve
 	// presubmits dynamically which we need for the PR history page.
 	var githubClient deckGitHubClient
-	var gitClient *git.Client
+	var gitClient git.ClientFactory
```

Comment:
Generally I prefer to have distinct PRs for the implementation and the integration, if possible. It has the advantage that its easier to review and makes me as a reviewer feel less "I really must not miss something". Also, with the combined PR if we have to revert the feature as a whole gets removed again rather than just the integration.